src/Optimizers/Hyper.jl

```diff
@@ -0,0 +1,108 @@
+using SparseArrays
+using KaHyPar
+
+@kwdef struct HyperGraphPartitioning <: Optimizer
+    metric::Function = removedsize
+    outer::Bool = false
+end
+
+
+function get_hypergraph(path)
+    all_indices = inds(path) # Assume unique indices across all tensors
```

Comment:
What do you mean by "assume unique indices across all tensors"?

Also, `inds` returns all the involved indices, *even the ones already summed in child `EinExpr`s*. I think you're looking for the following:
```suggestion
    # assume unique indices across all tensors
    all_indices = mapreduce(head, ∪, path)
```

Reply:
> What do you mean by "assume unique indices across all tensors"?

I meant that we want just the unique indices, so we do not repeat them when they appear to more than one tensor.


> Also, `inds` returns all the involved indices, _even the ones already summed in child `EinExpr`s_. I think you're looking for the following:

Ah... This makes sense! I will try that, thank you @mofeing !

Reply:
Sorry, I had a typo: it has to be `path.args` not `path` alone.